Feature/from string converter

slagroom opened the pull request:
Creates automatic from-string conversion types

- Creates a FromStringConversion<T> that looks for implied conversions from string in the T type metadata.

- Creates a FromStringConverter whose To<U> initializes a FromStringConversion to perform the conversion.

- Adds a missing null-check and test to the MapConverter type.

Commit:
Create FromStringConversion

- Creates FromStringConversion type for automatic conversion from
strings to supported types. Supported types are types that have either a
string constructor or a static Parse method that returns an instance of
the defining type.

Commit:
Create FromStringConverter

- Creates FromStringConverter; a type that implements an
IConverter<string> using instances of FromStringConversion.

Commit:
Add null-check and test to MapConverter

- Adds a test that null from causes MapConverter.To to throw an ANE and
adds the check explicitly to the code. Existing implementations of
IConversion throw ANE, but if a 3rd party implementation failed to
handle it then MapConverter would rethrow the NRE so the check was
added.

Insomniak47 commented on wimm.Missionary/FromStringConversion.cs:
```diff
@@ -0,0 +1,60 @@
+            if (parseMethod != null && parseMethod.ReturnType == typeof(T))
+                return str => (T)parseMethod.Invoke(null, new[] { str });
+
+            return null;
```
![image](https://user-images.githubusercontent.com/7606469/46265281-59f71300-c4fb-11e8-94de-de0776fbebb0.png)

slagroom replied:
Once it's feature complete for 1.0 I'll publish Secundatives and Maybeify this whole thing.

slagroom replied:
bump

Commit:
Use Maybe<T> instead of null

- Use Maybe<T> instead of null. Comparing to Maybe<SomeType<More,Types>>
is cumbersome, perhaps Exists should be exposed on Maybe<T> after all.

Commit:
Update to Secundatives 1.1 for Maybe<T>.Exists
